administration/src/components/Navigation.tsx

```diff
@@ -1,24 +1,40 @@
-import React from "react";
+import React, {useContext} from "react";
 import {Button, Navbar} from "@blueprintjs/core";
 import {NavLink} from "react-router-dom";
 import {Alignment} from "@blueprintjs/core/lib/esm/common/alignment";
-import RegionSelector from "./RegionSelector";
+import {RegionContext} from "../RegionProvider";
+import {AuthContext} from "../AuthProvider";
+import {Role} from "../generated/graphql";
 
 interface Props {
     onSignOut: () => void
 }
 
 const Navigation = (props: Props) => {
+    const region = useContext(RegionContext)
+    const [authContextData,] = useContext(AuthContext)
+    const role = authContextData?.administrator.role
```

Comment:
Could be refactored using a hook.

Reply:
Not sure whether that increases or decreases readability here for those two lines

Reply:
I've refactored the AuthContext a bit so that it's more readable.